orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pollers/EphemeralServerGroupsPoller.java

```diff
@@ -128,17 +135,25 @@ protected void tick() {
         Map<String, Object> cleanupOperation = buildCleanupOperation(ephemeralServerGroupTag, jobs);
         log.info((String) cleanupOperation.get("name"));
 
-        User systemUser = new User();
-        systemUser.setUsername("spinnaker");
-        systemUser.setAllowedAccounts(Collections.singletonList(ephemeralServerGroupTag.account));
-
-        AuthenticatedRequest.propagate(
-                () ->
-                    executionLauncher.start(
-                        ExecutionType.ORCHESTRATION,
-                        objectMapper.writeValueAsString(cleanupOperation)),
-                systemUser)
-            .call();
+        String taskUser =
+            getApplication(ephemeralServerGroupTag.application)
+                .map(it -> it.email)
+                .orElseGet(
+                    () -> {
+                      log.warn(
+                          "Failed to find application owner for server group '{}', "
+                              + "will use static 'spinnaker' user as fallback.",
+                          ephemeralServerGroupTag.serverGroup);
+                      return "spinnaker";
+                    });
+
+        AuthenticatedRequest.runAs(
```

Comment:
the opinion that the app owner is always a member of the app WRITE permissions is something we have implemented internally but isn't generally the case

also the allowedAccounts bit here is used as a fallback, but if we successfully look up the app owner as a service account then this would run with that service account's permissions. hopefully it succeeds, and we can via configuration make it so, but we may see failures for service accounts that are not set up to access restricted accounts (hopefully those become less of a thing as people start using those service accounts for managed delivery)

trying to think if I can offer any suggestions aside from just raising potential problems.. heh

Reply:
I doubt anyone else is using entity tags right, should we just consider this Netflix-only?

I suppose an alternative would be to look up the `executionId` and grab the authorization out of it ... at this point I don't know if we'd be splitting hairs tho.

Reply:
Look up the `executionId` of what? This is a poller, so there is no parent execution that we can lookup.

I will make the default for this task configurable, rather than `"spinnaker"`. That should check the box of many of the issues Cam brought up... in case anyone actually uses entity tags out there.

Good thoughts, thanks.

Reply:
Look up the `executionId` of what? This is a poller, so there is no parent execution that we can lookup.

I will make the default for this task configurable, rather than `"spinnaker"`. That should check the box of many of the issues Cam brought up... in case anyone actually uses entity tags out there.

Good thoughts, thanks.

Reply:
Updated. A configuration property will be preferred, but if unset, the original logic of using the application user will be used, still falling back to `spinnaker` if still nothing is found.

Reply:
The entity tag has the original execution id ... configuration property is reasonable to me tho.